File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static int HOURLY_PRODUCTION_RATE = 221;
    
    public double productionRatePerHour(int speed) {
        double successRate = 0;
        
        if (speed > 0 && speed <= 4) {
            successRate = 1.0;
        }
        else if (speed <= 8) {
            successRate = 0.9;
        }
        else if (speed == 9) {
            successRate = 0.8;
        }
        else if (speed == 10) {
            successRate = 0.77;
        }
        return HOURLY_PRODUCTION_RATE * speed * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)(productionRatePerHour(speed) / 60);
    }
}

File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed >= 1 && speed <= 4 then
        float(1)
    elif speed >= 5 && speed <= 8 then
        float(0.9)
    elif speed = 9 then
        float(0.8)
    elif speed = 10 then
        float(0.77)
    else
        float(0)

let productionRatePerHour (speed: int): float =
    float(221 * speed) * successRate(speed)

let workingItemsPerMinute (speed: int): int =
    int(productionRatePerHour(speed)/float(60))
